app/src/main/java/com/glowstudio/android/blindsjn/feature/login/LoginViewModel.kt

```
@@ -0,0 +1,130 @@
+                        phoneNumber = savedPhone,
+                        password = savedPassword,
+                        isLoading = true
+                    )
+                    try {
+                        val success = AuthRepository.login(context, savedPhone, savedPassword)
+                        if (success) {
+                            onLoginSuccess?.invoke(true)
+                        } else {
+                            _uiState.value = _uiState.value.copy(showInvalidCredentialsPopup = true)
+                        }
+                    } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)